subsys/bluetooth/Kconfig

```diff
@@ -131,6 +131,26 @@ config BT_EXT_ADV_MAX_ADV_SET
 	  Maximum number of simultaneous Bluetooth advertising sets
 	  supported.
 
+config BT_PER_ADV
+	bool "Periodic Advertising and Scanning support [EXPERIMENTAL]"
+	help
+	  Select this to enable Periodic Advertising API support..
+
```

Comment:
The Periodic Advertisements extends the extended advertisement sets, and thus the number of periodic advertisement sets is by `BT_EXT_ADV_MAX_ADV_SET`.

Reply:
If an application wants to have one connectable advertising and one periodic advertising at anytime, BT_EXT_ADV_MAX_ADV_SET  will be 2. But if we have an explicit Kconfig in Zephyr to restrict resources to one periodic advertising, it would save some RAM for the application.

Reply:
I'm not completely sure I follow. Are you suggesting that we have additional configs to control how many of the ADV sets may be used for periodic adv to save RAM? 

Enabling periodic adv support (`CONFIG_BT_PER_ADV`) doesn't increase the RAM, as all it adds are some functions and use some of the free bits in the flags. In other words, whether you want to use 1, 2 or 3 periodic advertisement doesn't really matter in terms of RAM (at least on the host side).

Reply:
@cvinayak I think this should be a controller Kconfig, since the Host cannot know how many of the ext adv sets support periodic adv:

>If the Advertising_Handle does not identify an advertising set that is already
configured for periodic advertising and the Controller is unable to support more
periodic advertising at present, the Controller shall return the error code
Memory Capacity Exceeded (0x07).

Reply:
@carlescufi  periodic adv set count is static Kconfig configuration, is handled in controller. https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/Kconfig.ll_sw_split#L159

https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c#L89

@joerchan The controller is not designed to share across allocation types/use heap. But yes, I understand. My original question was, how is host supporting a dynamic value? Can users be able to restrict and conserve memory.

Reply:
The host does not keep a lot of context information regarding the advertiser. This PR does not increase the size of the ext_adv object.
The host simply tries to create a new advertiser, or periodic advertiser, and if it fails, an error is reported to the application.